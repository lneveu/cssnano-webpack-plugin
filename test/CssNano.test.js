import MiniCssExtractPlugin from 'mini-css-extract-plugin';

import CssnanoPlugin from '../src/index';

import { createCompiler, compile } from './compiler';

import { readAsset, normalizedSourceMap } from './helpers';

describe('CssnanoPlugin', () => {
  jest.setTimeout(30000);

  it('should work with assets using querystring', () => {
    const config = {
      devtool: 'source-map',
      entry: {
        entry: `${__dirname}/fixtures/foo.css`,
      },
      plugins: [
        new MiniCssExtractPlugin({
          filename: '[name].css?v=test',
          chunkFilename: '[id].[name].css?v=test',
        }),
      ],
    };

    const compiler = createCompiler(config);
    new CssnanoPlugin({
      sourceMap: true,
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      expect(stats.compilation.errors).toEqual([]);
      expect(stats.compilation.warnings).toEqual([]);

      // eslint-disable-next-line guard-for-in
      for (const file in stats.compilation.assets) {
        // eslint-disable-next-line no-continue

        if (/\.css\?/.test(file)) {
          expect(readAsset(file, compiler, stats)).toMatchSnapshot(file);
        }

        // eslint-disable-next-line no-continue
        if (!/\.css.map/.test(file)) continue;
        expect(
          normalizedSourceMap(readAsset(file, compiler, stats))
        ).toMatchSnapshot(file);
      }
    });
  });
});
