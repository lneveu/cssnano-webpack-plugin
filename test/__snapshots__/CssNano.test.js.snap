// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssnanoPlugin should work with assets using querystring: entry.css.map?v=test 1`] = `"{\\"version\\":3,\\"sources\\": [replaced for tests], \\"names\\":[],\\"mappings\\":\\"AAAA,KACA,SACA,CACA,EACA,UACA\\",\\"file\\":\\"entry.css?v=test\\",\\"sourcesContent\\":[\\"body {\\\\n  color: red;\\\\n}\\\\na {\\\\n  color: blue;\\\\n}\\"],\\"sourceRoot\\":\\"\\"}"`;

exports[`CssnanoPlugin should work with assets using querystring: entry.css?v=test 1`] = `
"body{color:red}a{color:#00f}
/*# sourceMappingURL=entry.css.map?v=test*/"
`;
